### DISCUSSION TITLE
Need help which course to follow

### DISCUSSION THREAD

**Discussion by mariopetkoski** [Cyfrin/foundry-full-course-cu#522] [General]
I was 8 hours into the js-solidity course (the one released in 2022) and found that that this one just released. So my question is whether I should continue what I started or shift to this one. I wouldn't do both because combined they are nearly 60 hours. I need advice from someone experienced or someone that has done both.

**Comment by shikhar229169** [comment]
The thing is that both have their pros and cons.
Sometimes I feel like I should work with hardhat and sometimes want to work in foundry.

The thing is that you can do much more programmatic stuff with js but you can't do that in foundry.
But sometimes due to package and dependency issue it creates a lot of problem and it becomes harder to debug.

What you can do is watch the simple storage with foundry framework and make your own decision.

**Comment by theirrationalone** [comment]
If You wanna be a WEB3 Full Stack Dev, and you're Good In/Comfortable with/Love In, with Javascript then take this course: [Patrick's HH 2022 WEB3 Full Stack Javascript](https://www.youtube.com/watch?v=gyMwXuJrbJQ&t=53777s)

If You also wanna be a WEB3 Full Stack Dev and you're Good In/Comfortable with/Love In, with Python then take this course: [Patrick's HH 2021 WEB3 Full Stack Python](https://www.youtube.com/watch?v=M576WGiDBdQ)

And And And, If you wanna be a Solidity Smart Contracts Blockchain Developer, Looking to be a Master of, Blockchain Engineer, Security Expert,  Smart Contracts auditor, Truest Blockchain Professional only, then i'ld recommend you to take this course: [Patrick's Blockchain Dev](https://www.youtube.com/watch?v=umepbfKp5rI&list=PL4Rj_WH6yLgWe7TxankiqkrkVKXIwOP42), Please keep in mind, it's just an introduction + advanced blockchain development only. Blockchain security course is yet to be released. Follow @PatrickAlphaC to get updates about.

Good Luck 🍀for this Great Journey ahead.... 

Thank You! 🙏🙏🙏
